### PR TITLE
fix(@clayui/navigation): Adds missing `otherProps` for Vertical Navigation

### PR DESCRIPTION
**Pull request by diegonvs**

### CODE REVIEW

File: packages/clay-navigation/src/Vertical.tsx [hunk 1]
```diff
@@ -21,6 +21,11 @@ interface IItem {
 	 */
 	onClick?: () => void;
 
+	/**
+	 * Object of props for an item.
+	 */
+	itemProps?: Object;
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/Vertical.tsx]: Why an exclusive prop? where the element is the parent of the component.

File: packages/clay-navigation/src/Vertical.tsx [hunk 2]
```diff
@@ -154,6 +160,7 @@ export const ClayVerticalNav: React.FunctionComponent<IProps> = ({
 	items,
 	large,
 	spritemap,
+	...otherProps
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/Vertical.tsx]: The `IProps` interface needs to extend from the `HTMLElement` interface, it looks like we do not have an `HTMLNavElement`.

File: packages/clay-navigation/src/Vertical.tsx (hunk 1 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Vertical.tsx]: Solved after some internal conversation!

File: packages/clay-navigation/src/Vertical.tsx (hunk 2 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Vertical.tsx]: Solved after some internal conversation!

**Pull request merged**